src/item.h

```diff
@@ -1194,25 +1194,47 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
         item * gunmod_find( const itype_id& mod );
         const item * gunmod_find( const itype_id& mod ) const;
 
-        /**
-         * Returns currently active auxiliary (@ref is_auxiliary_gunmod) gun mod item.
-         * May return null if there is no such gun mod or if the gun is not in the
-         * auxiliary mode (@ref is_in_auxiliary_mode).
-         */
-        item * gunmod_current();
-        item const * gunmod_current() const;
-
         /*
          * Checks if mod can be applied to this item considering any current state (jammed, loaded etc.)
          * @param alert whether to display message describing reason for any incompatibility
          * @param effects whether temporary efects (jammed, loaded etc) are considered when checking
          */
         bool gunmod_compatible( const item& mod, bool alert = true, bool effects = true ) const;
 
-        /**
-         * Burst size (see ranged.cpp), includes effects from installed gunmods.
-         */
-        int burst_size() const;
+        struct gun_mode {
+            std::string mode; /** name of this mode */
```

Comment:
`id` would imply the string was the key which it's not
`name` would be ambiguous with the base `item`